src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -107,6 +107,9 @@ class GenericThreadStackManagerImpl_OpenThread
     CHIP_ERROR SEDUpdateMode();
     static void RequestSEDModeUpdate(chip::System::Layer * apSystemLayer, void * apAppState);
 #endif
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
It seems to me, that you want to split the SED and ICD cases in Thread platform. I wonder whether it's a good idea, as from the Thread platform perspective it should be transparent. No matter what heuristic is above, the Thread platform should care only about polling period updates. The rest depends on how the upper layers will use the platform API.

Reply:
As mentioned in my other comment, ICD will replace all SED calls. All #ifdef related to SED will be removed and only ICD will be used.

In terms of OpenThread stack. Nothing changes, Thread doesn't care about how Matter manages an ICD. It will be the same at the low level, (be a sleepy MTD device, set your polling and/or csl timing). 
However now ICD will be in full control of when we are in Active mode or when we can go in idle.

Reply:
Ok, great, my only concern was splitting ICD and SED functionalities. Once I'm sure it's only intermediate state before merging other PRs, I'm ok with that.